packages/caip/src/typeGuards.ts

```diff
@@ -0,0 +1,97 @@
+import { AssetId, AssetNamespace, AssetReference } from './assetId/assetId'
+import { ChainId, ChainNamespace, ChainReference } from './chainId/chainId'
+import {
+  ASSET_NAMESPACE_STRINGS,
+  ASSET_REFERENCE,
+  CHAIN_NAMESPACE,
+  CHAIN_REFERENCE
+} from './constants'
+import { isValidChainPartsPair, parseAssetIdRegExp } from './utils'
+
+export const isChainNamespace = (
+  maybeChainNamespace: ChainNamespace | string
+): maybeChainNamespace is ChainNamespace =>
+  Object.values(CHAIN_NAMESPACE).some((s) => s === maybeChainNamespace)
+
+export const isChainReference = (
+  maybeChainReference: ChainReference | string
+): maybeChainReference is ChainReference =>
+  Object.values(CHAIN_REFERENCE).some((s) => s === maybeChainReference)
+
+export const isAssetNamespace = (
+  maybeAssetNamespace: AssetNamespace | string
+): maybeAssetNamespace is AssetNamespace =>
+  ASSET_NAMESPACE_STRINGS.some((s) => s === maybeAssetNamespace)
+
+export const isAssetReference = (
+  maybeAssetReference: AssetReference | string
+): maybeAssetReference is AssetReference =>
+  Object.values(ASSET_REFERENCE).some((s) => s === maybeAssetReference)
+
+export const isAssetId = (maybeAssetId: AssetId | string): maybeAssetId is AssetReference => {
+  const matches = parseAssetIdRegExp.exec(maybeAssetId) ?? []
+  return (
+    isChainNamespace(matches[1]) && isChainReference(matches[2]) && isAssetNamespace(matches[3])
```

Comment:
is there a way we can use named capture groups here? obviously the ordering here is realistically not going to change but would be nice to be able to do `matches['chainReference']`

Reply:
There is! https://github.com/shapeshift/lib/pull/676/commits/ad56ace6ef4bc1914b2ede3bd18440ca3186f782